### DISCUSSION TITLE
Model implementation

### DISCUSSION THREAD

**Discussion by gavinny** [NASA-GISS/freshwater-forcing-workshop#5] [Model implementation]
This is the discussion on how climate models can/should implement the input of freshwater fluxes into the ocean.

**Comment by gavinny** [comment]
Consensus thoughts on the model implementation:

Different approaches needed for models that fix discharge at PI levels (Type-1) vs. models that maintain ice sheet mass balance (Type-2). Consequences are possibly different for global mass balance and ocean freshwater sources. 

Commonalities: 
Separation of spatial and time-varying parts. 
Regionalisation of fluxes to large drainage basins. Maybe 5 to 7 based on existing practice.
Spatial domain of iceberg melt and depth melt profiles can be derived from iceberg resolving models (Merino et al, 2016, Moon, Enderlin et al)
Depth profiles can be provided for the basal runoff/melt terms taken from fjord/calving front modeling.
Pre-industrial controls stay the same (ie mass balance is assumed). 
Annual resolution is sufficient (need to fill 1850-2021, with extensions to 2023 desirable). Link to CMIP7 protocol.
Simplifications can be to only use spatial map for icebergs (not depth), amalgamate regions per hemisphere, smooth timeseries. Groups can also pick and choose regions or to use iceberg flux or not depending on model capabilities. 

Suggestion that climatological fluxes by treated the same way as the anomalous flux - which may involve some new coding for existing models. This is not essential, but makes dealing with possibly negative FW anomalies easier (since the total discharge is never going to be negative). 

In historical period: Estimates of mass imbalance from multiple sources (see Bamber, Davison notes), including floating ice shelf variations and trends (more important in Antarctica). Possible use of ensemble mean calibrated output from Coulon et al (1950 onwards) as estimate of forced trend?

Scenarios from 2022 - for the time being, derived from existing ISMIP6 or other single model efforts (Coulon et al, Edwards group etc., UKESM?). Reconciliation with historical obs will be needed. Secondary priority. Note that anomalous flux in scenarios can be of both signs (i.e. net gain of mass in Antarctica for instance). 

Implementation:

Type 1 approach:

These models will have imbalances in the historical runs due to changes in the SMB on the ice sheets - for instance an increase in precipitation due to rising water vapor and convergence over Antarctica. Adding the anomalous flux to the PIControl discharge will provide a more accurate discharge into the Southern Ocean, but the implicit mass balance on the ice sheets will be different to observed (or simulated by the ISM), and so the overall ocean salinity/sea level will not necessarily match observations. 

Type 2 approach: 

These models assume mass balance of the ice sheets, via a change in iceberg/runoff flux (however implemented) on either instantaneous or decadal scale. 
Since there is an assumed balance, addition of the Anomalous FW terms will make the implicit mass balance of the ice sheets match observations and consequently the ocean salinity/sea level change will match expectations. However, the total discharge into the oceans may be different from observations depending on the accuracy of the resolved net integrated SMB. 

Assessment of the errors associated with each approach should be included in any paper.

**Comment by nicojourdain** [comment]
For type-2 models that remain type-2 throughout projections, we could discuss what to do with the freshwater anomaly that compensates future precipitation anomalies over the ice sheet. If the aim is really to close a global budget, then the freshwater anomaly could be spread globally, not under the form of a physical glacial freshwater flux around the ice sheets (which has strong impacts on sea ice). Then we can prescribe historical freshwater fluxes on top of this in the way we discussed at this meeting, without adding too unrealistic freshwater near ice sheets.

**Reply by swartn** [reply]
The climatological fluxes are pretty big - like 1500 to 3000 gt/yr around Antarctica (i.e. similar to observed calving/melt flux). This is because the "runoff" flux conceptually represents an implicit balance between surface accumulation and calving/melt. I'm not sure spreading this globally would be appropriate - and certainly it would have big impacts on the local salinity budget. As the models warm, and P-E increases, the runoff increases (in CanESM5, by about 200 Gt/yr).

The observed "anomalous" (i.e. non climatological flux) over the historical period is maybe 600 Gt/yr (what we put together from obs. papers for [SOFIA](https://egusphere.copernicus.org/preprints/2023/egusphere-2023-198/egusphere-2023-198.pdf)).

**Reply by nicojourdain** [reply]
@swartn : yes, the balance in preindustrial conditions should provide the background freshwater flux which has physical reasons to be applied as iceberg and ice-shelf melt (and surface runoff for Greenland). But the precipitation anomaly in the future should be stored for centuries over the ice sheet and should not go back directly into the ocean, especially if we add another ISMIP6-like perturbation on top of it.

In more details, I would run the climate spin-up as Type-2, save the freshwater flux that keeps the ice sheet mass constant and then apply it as background for all the simulations. If the model is switched to Type-1, the ocean volume/mass will increase in the historical and future scenarios. If it is switched to Type-2, the correction to conserve everything has nothing to do with the actual ice sheet dynamics, so I would just conserve the ice sheet mass, accounting for the historical/scenario perturbation discussed in this workshop and the precipitation anomaly over the ice sheet, and spread the corresponding freshwater globally.

**Reply by swartn** [reply]
Thanks @nicojourdain - I think this makes sense to me conceptually - sorry I think I misunderstood that you proposed to spread the whole flux, not just the anomaly part. From a practical standpoint of implementation across modelling centres, it would be interesting to see. From my perspective, some things I would wonder about would be, is there a potential to introduce a drift, and for which experiments to apply the constant background/anomaly spreading - only those with specified anomalous FW forcing, or for every experiment even if no additional FW forcing anomaly is applied?

**Reply by gavinny** [reply]
I think it might be useful to think about what the biases in the ESM might be due to as we consider how to work around them. For instance, excessive precipitation on the AIS could be due to too diffusive water vapor transport up on to the plateau. In the real world, this vapor rains out before it gets there. So a fix that puts the excess precip (back) in the oceans around the continent might actually produce a FW budget in the Southern Ocean that was more realistic, not less. If WV transports at 60S (say) are realistic, but it's just where the precip ends up that's the problem, this might all be finr.

**Reply by QianLi-Ocean** [reply]
Is there any feedback between the Anomalous FW and precipitation around the ice sheets? After adding a large amount of Anomalous FW, due to atmospheric cooling, the P-E around the ice sheets will decrease in future projections.

**Reply by gavinny** [reply]
in our runs it was small ~30 Gt/yr and not highly significant.

**Reply by QianLi-Ocean** [reply]
What amount of Anomalous FW does this correspond to? About 5%?

**Comment by gavinny** [comment]
This could be informed by an analysis of how large these fluxes actually are - it may well be that the net discharge with this approach is still within the uncertainties of the observed discharge - in which case it's probably not too bad.

**Comment by therealpaulholland** [comment]
Very naive question that might be off topic - sorry if so.  Until the models have dynamic ice sheets implemented, do we as a group prefer type 1 or type 2?  And does the introduction of a FW flux field change our preference? 

I think that if the models don't have a FW flux field imposed then maybe we might prefer type 2, in that at least the ice sheet is interacting with the rest of the climate and there will be an increasing discharge in forced runs.  

But as soon as there is a FW flux field imposed I think maybe we might prefer type 1.  If there is a trend of increasing snowfall onto Antarctica and that causes a sea level drop/salinification in a model, well that makes physical sense, right, as part of the simulation?  It might not match observations, but we'd like to know that because it will cause us to think carefully about the projection from that model. 

Is the above reasonable do we think?  If we did have the chance to influence what the modelling centres do on this topic, actually the switch from type 1 to type 2 or vice versa might be an important outcome.  If there is a majority agreement on some of this we should probably say so.

**Reply by xylar** [reply]
This question is very much on topic and not at all naive.  It's quite central in my view.

My sense from our discussions was that this community doesn't have a consensus on which we prefer.  For example, I strongly favor type 1 (which focuses on getting the right magnitude of freshwater fluxes around the ice sheets), whereas it seemed like @gavinny strongly preferred type 2 (which focuses on getting the right global change in salinity).

I would hope that more of a consensus will develop as part of starting to write up a protocol paper, but we will see.  It may be that we simply try to accommodate both types of models and lay out the pros and cons of each in the paper.

**Reply by therealpaulholland** [reply]
Thanks Xylar.  So do you agree that the implementation of a FW flux field could change the balance of pros and cons, perhaps?  Maybe that is an argument that would bring people around to your point of view.

**Reply by xylar** [reply]
@therealpaulholland, yes, I should have responded to that point.

I don't think type 1 models are really even worth considering in a situation where we are not trying to prescribe freshwater fluxes (whether in total or as an anomaly).  So this is why basically all ESMs are currently type 2 models.

But as soon as we start to prescribe FW fluxes or anomalies, that's where type 1 models make more sense to me than type 2.

**Reply by xylar** [reply]
That's also why the type 2 approach almost certainly make more sense for preindustrial spin-ups, and we will need to think about sensible ways for type 2 spinups to transition to type 1 historical runs.

**Reply by xylar** [reply]
@therealpaulholland, thanks for pointing out that I had switched type 1 and type 2 in my last post.  Fixed now.

**Reply by gavinny** [reply]
The survey we did of ~15 model groups - made it clear that the Type 2 approach is dominant, with a few models adopting Type 1, and a few groups doing nothing. I think we can (as a group) stay neutral while pointing out the consequences of each approach. We obviously have different personal preferences, but I don't know that we need to combine advocacy for one approach or the other with the building and reporting on the dataset and how it might be implemented.

**Reply by therealpaulholland** [reply]
I suppose this will all come out in the wash in the actual wording of the paper.  The distinction between 'pointing out consequences' and 'advocacy' is going to be a deliciously grey area. I look forward to reading the arguments either way!

**Reply by swartn** [reply]
I think that survey @gavinny conducted has some key information, that is quite hard to find otherwise. If the more detailed responses are available it would be great to be able to review these.

From discussions with Max, I roughly understood that in CMIP6, HadGEM had done something like this. That is, run as type2 during piControl, and determined the climatological calving/basal melt, and then held this rate fixed (i.e. type 1), during the historical. I may be off about the details (do you know @therealpaulholland? or maybe @andrewpauling?).

Either way, there is some kind of a divide between models with different capabilities. If your model has iceshelf cavities and explicit calving (but not an interactive ISM), I think you almost have to go to type 1 in forced runs, because otherwise things could get very out of hand, in the situation where you have strong forcing, but no ability to reduce iceshelf area. There is a related problem for models without explicit cavities/calving, which can possibly experience infinite surface melt, but this is less extreme.

**Reply by andrewpauling** [reply]
The calving/basal melt in the historical runs in HadGEM is held fixed at a level needed to balance the climatological SMB in the piControl. I am not sure what was done in the piControl though. It is not documented in the paper for the CMIP6 piControl runs.

**Reply by nicojourdain** [reply]
These questions are also important for ice-sheet coupling to climate models as I don't see any difference between having some mass stored in an ice sheet model and some mass stored in a file that would track cumulated prescribed fluxes.

**Reply by therealpaulholland** [reply]
@swartn nice points as ever, but I am feeling slow this morning after a pub trip and I don't follow your third paragraph above.  Are you saying that in models with ice shelf melting and iceberg input, you are necessarily inputting FW over a small/constrained area, and so large future changes in a type 2 approach could lead to large local FW fluxes and model instability?

IF that is the argument THEN 
  There is an implication for our present endeavour.  If we advocate a FW flux spatial field that is highly localised on the regions of real anomalous FW input, we will be causing problems for the type 2 models.  So then there would be a tension between those of us advocating for a very localised FW input, implying a type 1 approach, and those happy with a wider input, permitting a type 2 approach.
ELSE 
  please clarify 
END

**Reply by gavinny** [reply]
I think Neil's point is that if you have an ice shelf cavity that calculates basal melting, but no way of reducing the size of that cavity in a strong melt scenario, you could end up with an unbounded source of FW into the ocean.

**Reply by swartn** [reply]
Yes, what @gavinny said.

**Reply by therealpaulholland** [reply]
Ah, I think I see what you are getting at.  For example, if the Ross Ice Shelf cavity floods with warm water, the melt rate goes up massively.  In a coupled ISM the ice shelf will thin, and hence melting will drop because the thinner ice shelf is not sitting in the warm water.  In a static ice shelf the melting will stay elevated.  In the UKESM that stabilising feedback due to ice thinning is only just kicking in at 2100 in SSP5-8.5 (Figure 7b in https://doi.org/10.5194/tc-16-4053-2022). 

This is not true in general though.  E.g. there is good evidence that in the Amundsen Sea, the ice thinning and retreat leads to an increase in melting.  
https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2023GL103088
https://tc.copernicus.org/preprints/tc-2023-77/
https://doi.org/10.5194/egusphere-2023-1587
So then a static non-retreating ice shelf cavity would actually have a lower FW flux than would arise from a coupled ISM.

**Reply by swartn** [reply]
@therealpaulholland, yes I'm thinking of that longer timescale-e.g. ISMIP6 results, which consistently shows melt increases under ssp585, but plateau's towards 2100. The reason given is that all the ice shelves have disappeared. I think Helene's talk showed it here.  Sounds roughly consistent with what you are seeing in UKESM.

Interesting papers- thanks. Maybe my presumption is incorrect. I guess one would need to run a model like UKESM coupled and then with fixed cavities in a type 2 configuration and compare the resulting melt rates to test these ideas.

**Comment by dringeis** [comment]
Hi all, 

Following all the discussions above, I made some graphics to wrap my head around the 
 - type-1 and type-2 models
 - the implementation of these historical and scenario experiments
 - and the conservation of freshwater mass in the models' component interactions.

I thought it might be interesting to share this to make things clearer about what is going on in the different models and what we want to do for the experiment. Please let me know if you have some comments, if you think that I made a mistake or mixed things up. 

I think the historical experiment is quite straightforward, However, where I am not too sure, it is on the implementation for the scenario case (CMIP6 forcings, ...), see my question marks. 

Here is document with the four cases:
1.  Current situation
2. Historical experiment
3. Scenario experiment
4. and finally coupled models. 

[FW-workshop.pdf](https://github.com/NASA-GISS/freshwater-forcing-workshop/files/14337476/FW-workshop.pdf)

**Reply by rcullath** [reply]
These are excellent figures for discussion. The diagrams generally show surface runoff bypassing the ice sheet and being directly sent to the ocean. Note in ISMIP6, the ice sheet models were forced with SMB anomalies, which included runoff.

**Reply by xylar** [reply]
We discussed at the ISMIP7 ESM-ISM meeting on Friday that it would be great to have different groups make our own versions of your figures.  I think that would be really exciting!  @dringeis and @mankoff agreed to make versions of the figures that we can edit (e.g. in Powerpoint or Keynote).

**Reply by mankoff** [reply]
See https://github.com/NASA-GISS/freshwater-forcing-workshop/blob/main/freshwater.org for PlantUML versions, and link there to fig folder with PPT version https://github.com/NASA-GISS/freshwater-forcing-workshop/tree/main/fig

**Reply by dringeis** [reply]
Thanks! I will send an email to the ESM-ISM mailing list to say where the files are available.

**Reply by swartn** [reply]
Thanks @dringeis, diagrams are always helpful! 

I have a question about the equation setup for type 2 models. Currently you have 1+2+3=0. However, as I see it, 2 and 3 are not different fluxes, they are one and the same flux (unless I don't understand what you are drawing). For mass balance (i.e. type 2), what we enforce in models like canesm is that 1=3 (or equivalently 1=2). Am I missing something? (perhaps if 2 is meant to be an explicit calving/basal melt flux, in which case, 1+2+3=0, then the name for 2 should be changed to reflect this)

**Reply by dringeis** [reply]
Thanks Neil!

The way I see it, I think 2 and 3 are different fluxes:
- 2 is a physical surface runoff flux. Snow/firn is melting from radiation on the surface, and percolates through the snow/firn layers until it reaches the bottom of it and makes it way to the ocean as liquid FW flux through a river-like mecanism. The snow/firn model is of limited thickness, 3-15m, depending on the number of layers and personal model choices.
- 3 is a flux approximation designed to emulate ice sheet/shelves dynamics under the assumption that the ice sheet is in steady state and the accumulation/ablation on the surface is almost directly represented in the basal melt and iceberg calving fluxes. Maybe _implicit_ is not the good term here...
  - *__For more details:__ The top of the ice sheet is kept at a constant atmospheric height, the snow/firn model on top of it. The mass or thickness of the snow/firn model is kept somewhat constant and as we have accumulation of snow on the ice sheet high parts and ablation on the lower parts. These are added and subtracted respectively and accumulated and then send to the ocean as FW fluxes without or with a delay to represent some kind of ice sheet response time (I think we used 10 years in modelE). Note that the flux cannot be negative, no "iceberg sucking" allowed! 😄 There is also the question of where this flux is added in the ocean...* These details are surely very model-dependent.

This all a big approximation which is getting less and less true with climate change.

**Reply by swartn** [reply]
@dringeis - ok thanks for clarifying what you mean. I agree the explicit surface melt, and implicit accumulation/calving=FW flux can be separated. What led to my confusion, is that for us, and I understand quite a few other models, both of these fluxes are practically added into river runoff. Diagrammatically it might be useful to clarify.

**Reply by dringeis** [reply]
Oh ok, I see, interesting!

hmm I can try something, but I am a bit afraid that it would make the diagrams a bit too cluttered, because this is too model-dependent and I cannot represent all model flavors. This is the reason why I did not try to include the geographical distribution of the FW in the ocean to focus on the FW fluxes balances.

**Reply by xylar** [reply]
I think it would be best if different centers make their own versions of the figures, rather than making one that covers everyone.
